File: frontend/types/daily_summary.ts
export interface DailyProjectSummary {
  id: number;
  project_id: number;
  summary_date: string;
  summary: string;
  risk_summary: string;
  next_action: string;
  management_attention: string;
  created_at: string;
}

export interface ParsedAiAdvice {
  problem_summary?: string;
  possible_causes?: string[];
  checks?: string[];
  recommended_next_actions?: string[];
  suggested_participants?: string[];
  escalation_recommended?: boolean;
}

File: backend/app/schemas/daily_summary.py
"""Daily project summary API schemas."""

from __future__ import annotations

from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class DailySummaryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    summary_date: date
    summary: str
    risk_summary: str
    next_action: str
    management_attention: str
    created_at: datetime
